// @clickhouse/client/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.extractErrorAtTheEndOfChunk = exports.EXCEPTION_TAG_HEADER_NAME = exports.defaultJSONHandling = exports.ClickHouseSpanKind = exports.ClickHouseSpanStatusCode = exports.ClickHouseSpanNames = exports.TupleParam = exports.isException = exports.isRow = exports.isProgressRow = exports.SimpleColumnTypes = exports.parseColumnType = exports.RecordsJSONFormats = exports.SingleDocumentJSONFormats = exports.StreamableJSONFormats = exports.StreamableFormats = exports.SupportedRawFormats = exports.SupportedJSONFormats = exports.SettingsMap = exports.ClickHouseLogLevel = exports.parseError = exports.ClickHouseError = exports.drainStream = exports.ResultSet = exports.createClient = exports.ClickHouseClient = void 0;
var client_1 = require("./client");
Object.defineProperty(exports, "ClickHouseClient", { enumerable: true, get: function () { return client_1.NodeClickHouseClient; } });
var client_2 = require("./client");
Object.defineProperty(exports, "createClient", { enumerable: true, get: function () { return client_2.createClient; } });
var result_set_1 = require("./result_set");
Object.defineProperty(exports, "ResultSet", { enumerable: true, get: function () { return result_set_1.ResultSet; } });
var stream_1 = require("./connection/stream");
Object.defineProperty(exports, "drainStream", { enumerable: true, get: function () { return stream_1.drainStream; } });
/**
 * Re-export common (formerly @clickhouse/client-common) runtime values.
 *
 * These are intentionally re-exported through local bindings (rather than a direct
 * `export { ... } from './common/index'`) so that the `@deprecated` JSDoc tags
 * applied to them in `./common/index` are NOT propagated to consumers of this package.
 * Importing these values from `@clickhouse/client` is the recommended, non-deprecated path.
 */
const index_1 = require("./common/index");
exports.ClickHouseError = index_1.ClickHouseError;
exports.parseError = index_1.parseError;
exports.ClickHouseLogLevel = index_1.ClickHouseLogLevel;
exports.SettingsMap = index_1.SettingsMap;
exports.SupportedJSONFormats = index_1.SupportedJSONFormats;
exports.SupportedRawFormats = index_1.SupportedRawFormats;
exports.StreamableFormats = index_1.StreamableFormats;
exports.StreamableJSONFormats = index_1.StreamableJSONFormats;
exports.SingleDocumentJSONFormats = index_1.SingleDocumentJSONFormats;
exports.RecordsJSONFormats = index_1.RecordsJSONFormats;
/** @deprecated Superseded by the `@clickhouse/datatype-parser` package (`parseDataType` + its `Node` AST); slated for removal in a future major version. */
exports.parseColumnType = index_1.parseColumnType;
/** @deprecated Superseded by the `@clickhouse/datatype-parser` package (`parseDataType` + its `Node` AST); slated for removal in a future major version. */
exports.SimpleColumnTypes = index_1.SimpleColumnTypes;
exports.isProgressRow = index_1.isProgressRow;
exports.isRow = index_1.isRow;
exports.isException = index_1.isException;
exports.TupleParam = index_1.TupleParam;
exports.ClickHouseSpanNames = index_1.ClickHouseSpanNames;
exports.ClickHouseSpanStatusCode = index_1.ClickHouseSpanStatusCode;
exports.ClickHouseSpanKind = index_1.ClickHouseSpanKind;
exports.defaultJSONHandling = index_1.defaultJSONHandling;
exports.EXCEPTION_TAG_HEADER_NAME = index_1.EXCEPTION_TAG_HEADER_NAME;
exports.extractErrorAtTheEndOfChunk = index_1.extractErrorAtTheEndOfChunk;
